felix/dataplane/linux/int_dataplane.go

```diff
@@ -2292,6 +2307,25 @@ func (d *InternalDataplane) apply() {
 		}(ipSets)
 	}
 
+	// Track if we need to perform additional map updates after tables have been programmed.
+	// This is because there is a bidirectional dependency between maps and rules.  We need to
+	// program maps in case any rule references them, and we also need to update map members which
+	// reference rules after the rules have been programmed.
```

Comment:
Could probably get rid of this is we moved the maps programming logic into the table struct instead of having a separate component doing it - then we'd program everything as a single transaction.

That's probably preferable to this, but this was easier to mock up.

Reply:
I coded up this modification and then accidentally checked out over the code :sob: 

Main thing is that order of operations in the transaction needs to be correct -

- Create table
- Create empty maps as needed
- Create rules
- Add map elements that reference rules

And cleanup of course would be the reverse.